### ISSUE TITLE
sslv3_client_method dont work 

### ISSUE THREAD

**Issue by budibeta** [openssl/openssl#16982]
deleted

**Comment by kaduk** [comment]
Why can't you just use `TLS_client_method()`?

**Comment by budibeta** [comment]
> Why can't you just use `TLS_client_method()`?

some websites cannot visited by tls 1.2, i write client, not a server

**Comment by mattcaswell** [comment]
> when i use sslv3_client_method on server that accept tlsv1.3, my client is still connection reset.

I assume you mean `SSLv3_client_method()`. This is a version fixed method that *only* talks SSLv3. It cannot talk TLSv1.3 and most servers will not support SSLv3 so a connection reset is not surprising.

> some websites cannot visited by tls 1.2, i write client, not a server

This makes no sense. `TLS_client_method()` is a version flexible method designed for use by clients that will negotiate the highest available SSL/TLS version that both the client and the server support. This is the method that your application should be using. All other client methods are deprecated.

**Comment by budibeta** [comment]
> > when i use sslv3_client_method on server that accept tlsv1.3, my client is still connection reset.
> 
> I assume you mean `SSLv3_client_method()`. This is a version fixed method that _only_ talks SSLv3. It cannot talk TLSv1.3 and most servers will not support SSLv3 so a connection reset is not surprising.
> 
> > some websites cannot visited by tls 1.2, i write client, not a server
> 
> This makes no sense. `TLS_client_method()` is a version flexible method designed for use by clients that will negotiate the highest available SSL/TLS version that both the client and the server support. This is the method that your application should be using. All other client methods are deprecated.

i have ever used both TLS_client_method and SSLv3_client_method, work on stackoverflow.com, google.com, github, openssl.org,

but dont work on some website like https://ouo.io/ and unknowncheats.me

i write ssl proxy server, work on tls 1.2 using SSLv3_client_method

but dont dont on every website that use tls 1.3

unknowncheats.me use tls 1.3, my proxy server throw err_connection_reset
then i rebuild the source and enabling sslv3, i can use SSLv3_client_method, when i made a test the result still same err_connection_reset
![Err-Connection-Reset-Error](https://user-images.githubusercontent.com/68463485/140612760-c7b933e0-45d1-47fc-b731-35fe76f14576.jpg)

<img width="505" alt="Screenshot_1" src="https://user-images.githubusercontent.com/68463485/140612700-bbbd7d04-7b63-48b9-9074-2ec32a267402.png">

**Comment by kroeckx** [comment]
If you write a client, you should always use TLS_client_method().
Do not use any other method like SSLv3_client_method().
SSLv3_client_method only supports SSLv3, it does not support
TLS 1.2 or 1.3. 

If you use SSLv3_client_method() and can connect to a server
that supports TLS 1.2, it means the server also still supports
SSLv3 and you're talking SSLv3 to it, not TLS 1.2.

SSLv3 is insecure. You should stop using it.

TLS_client_method() supports all SSL and TLS versions. It will
pick the highest version supported by both client and server.

Note that TLS_client_method() probably has SSLv3 disabled by
default, because it's insecure. If you need to be able to support
SSLv3, you will need to enable it.

In short: Use TLS_client_method().

**Comment by budibeta** [comment]
> If you write a client, you should always use TLS_client_method(). Do not use any other method like SSLv3_client_method(). SSLv3_client_method only supports SSLv3, it does not support TLS 1.2 or 1.3. If you use SSLv3_client_method() and can connect to a server that supports TLS 1.2, it means the server also still supports SSLv3 and you're talking SSLv3 to it, not TLS 1.2. SSLv3 is insecure. You should stop using it. TLS_client_method() supports all SSL and TLS versions. It will pick the highest version supported by both client and server. Note that TLS_client_method() probably has SSLv3 disabled by default, because it's insecure. If you need to be able to support SSLv3, you will need to enable it. In short: Use TLS_client_method().

youre think you smart enough, i said i was ever used **TLS_client_method** and the result is still same **ERR_CONNECTION_RESET** in chrome, when i visit unknowncheats.me

i have a proof, **TLS_client_method** and **SSLv23_client_method** is same !!!
this is the reference from ssl.h
<img width="303" alt="Screenshot_2" src="https://user-images.githubusercontent.com/68463485/140617124-4a66b874-6735-44ca-9908-c4933a08b0e2.png">

and TLS_client_method success on google, stackoverflow and many sites using my proxy, why i cannot visit unknowncheats.me using my proxy, that because difference ssl version, so i should change the TLS_client_method into SSLv3_client_method, but the result is same even i used SSLv3_client_method and enable sslv3

**Comment by kroeckx** [comment]
unknowncheats.me only supports TLS 1.2 and 1.3, it does not support SSLv3.
Other sides might still support SSLv3.

Yes, SSLv23_client_method and TLS_client_method are the same, but
SSLv23_client_method is a confusing name, it supports more than
just SSLv2 and SSLv3.

**Comment by budibeta** [comment]
> unknowncheats.me only supports TLS 1.2 and 1.3, it does not support SSLv3. Other sides might still support SSLv3. Yes, SSLv23_client_method and TLS_client_method are the same, but SSLv23_client_method is a confusing name, it supports more than just SSLv2 and SSLv3.

do you have any idea what should i do, so my proxy can connect unknowncheats.me, also SSL_free sometimes cause process crash, i used non-block socket in my proxy server
